### PR TITLE
NPC Talk Rework and Enchanting additions/fixes

### LINKED ISSUE #635: Enchant scrolls do not work
they just dont man
image

Socket scrolls are working as intended.

### LINKED ISSUE #636: Peachy crash when trying to enchant
Attempting to use Peachy disconnects the user from the server. Server does not crash.

Stack trace:
```
[Fatal] MapleServer2.Network.Session: Exception in recv PipeScheduler: System.NullReferenceException: Object reference not set to an instance of an object.
   at MapleServer2.PacketHandlers.Game.Helpers.EnchantHelper.HandleNpcTalkEventType(GameSession session, NpcScript npcScript, Int32 eventId) in C:\MS2-Server\MapleServer2\MapleServer2\PacketHandlers\Game\Helpers\EnchantHelper.cs:line 107
   at MapleServer2.PacketHandlers.Game.NpcTalkHandler.HandleEnchant(GameSession session, PacketReader packet) in C:\MS2-Server\MapleServer2\MapleServer2\PacketHandlers\Game\NpcTalkHandler.cs:line 415
   at MapleServer2.PacketHandlers.Game.NpcTalkHandler.Handle(GameSession session, PacketReader packet) in C:\MS2-Server\MapleServer2\MapleServer2\PacketHandlers\Game\NpcTalkHandler.cs:line 52
   at MapleServer2.Network.Session.ReadRecvPipe(PipeReader reader) in C:\MS2-Server\MapleServer2\MapleServer2\Network\Session.cs:line 257
```

### LINKED ISSUE #646: Attempting to change attributes on armor/accessory while the gear is equipped kicks you from the server
How to reproduce:
- Summon any armor/accessory with random stats
- Equip it
- Try to change the attributes using Ophelia
- The server console will throw this FTL exception:
image

This doesn't occur when the item is in your inventory.

### PR DESCRIPTION
**Pull request by Zintixx**
- scripts/quests are now parsed using `Maple2.File.Parser`
- Rework on NPC talk. 
  - Any additional recommendations is appreciated
- Implemented enchanting dialog for Peachy and Ophelia
- Implements NPC Job scripts. Note not all are implemented right now. Will come in later PR.
- Implements enchanting server notices if enchant level is 12 or above
- Added additional NPC lua scripts
- Fixed shops. Turns out there is no "uid". it's just shop id. Shop Items reflect this change
- Fixes #636 
- Fixes #635 
- Fixes #646

### CODE REVIEW

File: Maple2Storage/Scripts/Npcs/11000254.lua [hunk 1]
```diff
@@ -0,0 +1,18 @@
+-- Jane
+function handleGoto(nextScript)
+    local itemCount = ScriptManager.GetPlayer().GetItemCount(39000014);
+    print(itemCount)
```
Comment (AngeloTadeucci) [inline Maple2Storage/Scripts/Npcs/11000254.lua:4]:
```suggestion
```

File: Maple2Storage/Scripts/Npcs/11000508.lua [hunk 2]
```diff
@@ -0,0 +1,83 @@
+function getBeginEventId(itemRarity, enchantLevel)
+    if enchantLevel == 0 then
+        if itemRarity <= 4 then
+            return 1100
+        elseif itemRarity == 5 then
+            return 1000
+        elseif itemRarity == 6 then
+            return 1001
+        end
+    elseif enchantLevel <= 2 then
```
Comment (AngeloTadeucci) [inline Maple2Storage/Scripts/Npcs/11000508.lua:10]: `return 1998 + enchantLevel`?

File: Maple2Storage/Scripts/Npcs/11000510.lua [hunk 3]
```diff
@@ -0,0 +1,79 @@
+function getBeginEventId(itemRarity, enchantLevel)
+    if enchantLevel == 0 then
+        if itemRarity <= 4 then
+            return 1100
+        elseif itemRarity == 5 then
+            return 1000
+        elseif itemRarity == 6 then
+            return 1001
+        end
+    elseif enchantLevel <= 2 then
```
Comment (AngeloTadeucci) [inline Maple2Storage/Scripts/Npcs/11000510.lua:10]: same thing here `return 1998 + enchantLevel`?

File: Maple2Storage/Scripts/Npcs/11000994.lua [hunk 4]
```diff
@@ -0,0 +1,6 @@
+-- Lotachi
+function meetsJobScriptRequirement()
+    if ScriptManager.GetPlayer().HasQuestStarted(10001840) then
+        return true
+    end
```
Comment (AngeloTadeucci) [inline Maple2Storage/Scripts/Npcs/11000994.lua:5]:
```suggestion
    return ScriptManager.GetPlayer().HasQuestStarted(10001840)
```

File: MapleServer2/Database/Seeding/ShopItemsSeeding.sql [hunk 5]
```diff
@@ -24,1025 +24,1025 @@ CREATE TABLE `shop_items`
     `required_item_id`                 int      NOT NULL,
     `required_quest_alliance`          smallint NOT NULL,
     `sale_price`                       int      NOT NULL,
-    `shop_uid`                         int         DEFAULT NULL,
+    `shop_id`                         int         DEFAULT NULL,
```
Comment (darkvergus) [inline MapleServer2/Database/Seeding/ShopItemsSeeding.sql:27]: FIX TABBING MADGE

File: MapleServer2/PacketHandlers/Game/NpcTalkHandler.cs [hunk 6]
```diff
@@ -82,313 +89,481 @@ private static void HandleRespond(GameSession session, PacketReader packet)
...
+        npcTalk.DialogType = GetDialogType(nextScript, npcTalk);
+        responseSelection = GetResponseSelection(kind, npcTalk.DialogType, npcTalk.ContentIndex, nextScript);
+        if (nextScript != null)
         {
-            npcTalk.ContentIndex++;
+            npcTalk.TalkFunction(session, nextScript.Contents[npcTalk.ContentIndex].FunctionId, "preTalkActions");
         }
+        session.Send(NpcTalkPacket.ContinueChat(npcTalk.ScriptId, npcTalk.DialogType, responseSelection, npcTalk.ContentIndex, npcTalk.QuestId));
+    }
+    private static NpcScript GoToScript(GameSession session, int selectedIndex, NpcScript currentScript, NpcTalk npcTalk, ScriptMetadata metadata)
```
Comment (AngeloTadeucci) [inline MapleServer2/PacketHandlers/Game/NpcTalkHandler.cs:319]: new line before

File: MapleServer2/PacketHandlers/Game/NpcTalkHandler.cs [hunk 7]
```diff
@@ -82,313 +89,481 @@ private static void HandleRespond(GameSession session, PacketReader packet)
...
-        session.Send(NpcTalkPacket.ContinueChat(nextScriptId, responseType, dialogType, npcTalk.ContentIndex, npcTalk.QuestId));
-        // It appears if content has buttonset roulette, it's send again on every continue chat, unsure why since it doesn't break anything
+        return metadata.NpcScripts.FirstOrDefault(x => x.Id == currentScript.Contents[npcTalk.ContentIndex].Distractor[selectedIndex].Goto[0]);
     }
 
-    private static void HandleNextQuest(GameSession session, PacketReader packet)
+    private static DialogType GetDialogType(NpcScript npcScript, NpcTalk npcTalk)
     {
-        int questId = packet.ReadInt();
-        short mode = packet.ReadShort();
```
Comment (darkvergus) [inline MapleServer2/PacketHandlers/Game/NpcTalkHandler.cs:340]: extra space

File: MapleServer2/PacketHandlers/Game/NpcTalkHandler.cs (hunk 6 above)
Comment (darkvergus) [inline MapleServer2/PacketHandlers/Game/NpcTalkHandler.cs:319]: space before the function

File: MapleServer2/Types/BankInventory.cs [hunk 8]
```diff
@@ -214,10 +214,11 @@ public void Expand(GameSession session)
 
     public void LoadBank(GameSession session)
     {
+        session.Send(StorageInventoryPacket.Expand(ExtraSize));
         session.Send(StorageInventoryPacket.Update());
         session.Send(StorageInventoryPacket.Expand(ExtraSize));
```
Comment (AngeloTadeucci) [inline MapleServer2/Types/BankInventory.cs:219]: Expand twice? 🤔

**Review by AngeloTadeucci — CHANGES_REQUESTED** [review]

File: MapleServer2/Types/BankInventory.cs (hunk 8 above)
Comment (Zintixx) [reply MapleServer2/Types/BankInventory.cs:219]: its sent twice. not sure why

**Review by darkvergus — APPROVED** [review]

**Review by AngeloTadeucci — APPROVED** [review]

**Pull request merged**